tests/runtime/filter_parser.c

```diff
@@ -812,6 +812,86 @@ void flb_test_filter_parser_preserve_original_field()
     flb_destroy(ctx);
 }
 
+void flb_test_filter_parser_hash_value_field()
+{
+    int ret;
+    int bytes;
+    char *p, *output, *expected;
+    flb_ctx_t *ctx;
+    int in_ffd;
+    int out_ffd;
+    int filter_ffd;
+    struct flb_parser *parser;
+
+    struct flb_lib_out_cb cb;
+    cb.cb   = callback_test;
+    cb.data = NULL;
+
+    clear_output();
+
+    ctx = flb_create();
+
+    /* Configure service */
+    flb_service_set(ctx, "Flush", FLUSH_INTERVAL, "Grace", "1", "Log_Level", "debug", NULL);
+
+    /* Input */
+    in_ffd = flb_input(ctx, (char *) "lib", NULL);
+    TEST_CHECK(in_ffd >= 0);
+    flb_input_set(ctx, in_ffd,
+                  "Tag", "test",
+                  NULL);
+
+    /* Parser */
+    parser = flb_parser_create("dummy_test", "regex", "^(?<INT>[^ ]+) (?<FLOAT>[^ ]+) (?<BOOL>[^ ]+) (?<STRING>.+)$",
```

Comment:
`parser` should be allocated some of the amount of heap memory.
So, we need to teardown them.
To destroy the allocated region, we need to use `flb_parser_destroy` to clean up it before exiting this test case.

Reply:
Its not destroyed in other UTs. Also this results in a use after free bug(the parser memory is repurposed)